tests/v2/validation/observability/stackstate_test.go

```diff
@@ -0,0 +1,271 @@
+			ChartName:   charts.StackstateExtensionsName,
+			ReleaseName: charts.StackstateExtensionsName,
+			Version:     latestUIPluginVersion,
+		}
+
+		err = charts.InstallStackstateExtension(client, extensionOptions)
+		require.NoError(ss.T(), err)
+
+		log.Info("Adding stack state extension configuration.")
+
+		steveAdminClient, err := client.Steve.ProxyDownstream(localCluster)
+		require.NoError(ss.T(), err)
+
+		crdConfig := observability.NewStackstateCRDConfiguration(charts.StackstateNamespace, ss.stackstateConfigs)
+		crd, err := steveAdminClient.SteveType(charts.StackstateCRD).Create(crdConfig)
+		require.NoError(ss.T(), err)
+
+		_, err = steveAdminClient.SteveType(charts.StackstateCRD).ByID(crd.ID)
+		require.NoError(ss.T(), err)
+	}
+
+	latestSSVersion, err := ss.client.Catalog.GetLatestChartVersion(charts.StackstateK8sAgent, rancherPartnerCharts)
+	ss.stackstateAgentInstallOptions = &charts.InstallOptions{
+		Cluster:   cluster,
+		Version:   latestSSVersion,
+		ProjectID: project.ID,
+	}
+}
+
+func (ss *StackStateTestSuite) TestStackState() {
+	subSession := ss.session.NewSession()
+	defer subSession.Cleanup()
+
+	client, err := ss.client.WithSession(subSession)
+	require.NoError(ss.T(), err)
+
+	var stackstateConfigs observability.StackStateConfigs
+	config.LoadConfig(stackStateConfigFileKey, &stackstateConfigs)
+
+	initialStackstateAgent, err := extencharts.GetChartStatus(client, ss.cluster.ID, charts.StackstateNamespace, charts.StackstateK8sAgent)
+	require.NoError(ss.T(), err)
+
+	if !initialStackstateAgent.IsAlreadyInstalled {
+		log.Info("Installing stack state agent on the provided cluster")
+
+		systemProject, err := projects.GetProjectByName(client, ss.cluster.ID, systemProject)
+		require.NoError(ss.T(), err)
+		systemProjectID := strings.Split(systemProject.ID, ":")[1]
+
+		err = charts.InstallStackstateAgentChart(ss.client, ss.stackstateAgentInstallOptions, stackstateConfigs.ClusterApiKey, stackstateConfigs.Url, systemProjectID)
+		require.NoError(ss.T(), err)
+		log.Info("Stack state chart installed successfully")
+
+		ss.T().Log("Verifying the deployments of stackstate agent chart to have expected number of available replicas")
+		err = extencharts.WatchAndWaitDeployments(client, ss.cluster.ID, charts.StackstateNamespace, meta.ListOptions{})
+		require.NoError(ss.T(), err)
+
+		ss.T().Log("Verifying the daemonsets of stackstate agent chart to have expected number of available replicas nodes")
+		err = extencharts.WatchAndWaitDaemonSets(client, ss.cluster.ID, charts.StackstateNamespace, meta.ListOptions{})
+		require.NoError(ss.T(), err)
+	}
+}
+
+func (ss *StackStateTestSuite) TestUpgradeStackstateAgentChart() {
+	subSession := ss.session.NewSession()
+	defer subSession.Cleanup()
+
+	client, err := ss.client.WithSession(subSession)
+	require.NoError(ss.T(), err)
+
+	versionsList, err := client.Catalog.GetListChartVersions(charts.StackstateK8sAgent, rancherPartnerCharts)
+	require.NoError(ss.T(), err)
```

Comment:
curious if there's anything that could break if stackstate-server isn't on the same version as the stackstate-agent? this can cause issues with rancher, so it always has rancher-version == rancher-agent-version

Reply:
good question. I do not think they version locked based on the values.yaml but can confirm with the stackstate team.